From 5ec265deac0da077c9b1e23fc52abe1b5f0696b5 Mon Sep 17 00:00:00 2001
From: Leah Rowe <leah@libreboot.org>
Date: Sat, 13 Jan 2024 14:57:46 +0000
Subject: [PATCH 1/1] buildgcc: don't treat binutil warnings as errors

binutils 2.32 has too many build warnings on modern toolchains,
and newer gcc versions are much more pedantic about warnings,
treating them as errors by default.

instead of patching binutils like before, just let the warnings
persist. the warnings are benign. a user on gnuboot irc had serious
issues building binutils 2.32 specifically, on current gentoo as
of 13 January 2024. this patch mitigates those warning messages.

Signed-off-by: Leah Rowe <leah@libreboot.org>
---
 util/crossgcc/buildgcc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/crossgcc/buildgcc b/util/crossgcc/buildgcc
index 505cd7484c..6f1953e68d 100755
--- a/util/crossgcc/buildgcc
+++ b/util/crossgcc/buildgcc
@@ -719,7 +719,7 @@ build_BINUTILS() {
 		--disable-werror --disable-nls --enable-lto --enable-gold \
 		--enable-interwork --enable-multilib \
 		--enable-plugins --enable-multilibs \
-		CFLAGS="$HOSTCFLAGS" \
+		CFLAGS="$HOSTCFLAGS -Wno-error -w" \
 		CXXFLAGS="$HOSTCFLAGS" \
 		|| touch .failed
 	# shellcheck disable=SC2086
-- 
2.39.2

